playwright/_impl/_browser.py

```diff
@@ -145,6 +146,10 @@ async def new_page(
         return page
 
     async def close(self) -> None:
+        if self._is_connected_over_websocket:
```

Comment:
Do you want to move it to under _is_closed_or_closing = True? Also, why is logic different from the one in TS? Does TS close the browser on browser.close?

To me, the ideal workflow would be: `close` terminates the underlying connection, connection termination calls _on_close everywhere. That way, terminating connection due to networking issues would result in the same proper cleanup.

Reply:
move it below it.

TS does it via an event, see here in another place: https://github.com/microsoft/playwright/blob/fff1f3d45c243bec65c7e6c43e6fe5e36607dc59/src/client/browserType.ts#L163-L179